packages/dev/loaders/test/integration/babylon.sceneLoader.test.ts

```diff
@@ -197,10 +197,960 @@ describe("Babylon Scene Loader", function () {
+                            "materialMedium.isReady(meshes[0])": materialMedium!.isReady(meshes[0]!),
+                            "materialMedium.isReady(meshes[1])": materialMedium!.isReady(meshes[1]!),
+                            "materialHigh.isReady(meshes[0])": materialHigh!.isReady(meshes[0]!),
+                            "materialHigh.isReady(meshes[1])": materialHigh!.isReady(meshes[1]!),
+                        })
+                    );
+                });
+
+                return promise.then(() => window.scene!.whenReadyAsync()).then(() => Promise.all(promises));
+            });
+
+            expect(assertionData.length).toBeGreaterThanOrEqual(3);
+
+            assertionData.forEach((promise) => {
+                Object.keys(promise).forEach((key) => {
+                    expect(promise[key as keyof typeof promise], key).toBe(true);
+                });
+            });
+        });
+
+        it("Load LevelOfDetail with onMaterialLODsLoadedObservable", async () => {
+            const materialNames = ["Low", "Medium", "High"];
+            const assertionData = await page.evaluate((materialNames) => {
+                const promises = new Array<Promise<void>>();
+
+                const data: { [key: string]: string[] } = {};
+
+                BABYLON.SceneLoader.OnPluginActivatedObservable.addOnce((loader) => {
+                    const observer = (loader as GLTFFileLoader).onExtensionLoadedObservable.add((extension) => {
+                        if (extension instanceof (BABYLON.GLTF2 as any).Loader.Extensions.MSFT_lod) {
+                            (loader as GLTFFileLoader).onExtensionLoadedObservable.remove(observer);
+                            (extension as any).onMaterialLODsLoadedObservable.add((indexLOD: number) => {
+                                data[materialNames[indexLOD]] = [window.scene!.getMeshByName("node0")!.material!.name, window.scene!.getMeshByName("node1")!.material!.name];
+                            });
+                        }
+                    });
+
+                    promises.push((loader as GLTFFileLoader).whenCompleteAsync());
+                });
+
+                return BABYLON.SceneLoader.AppendAsync("https://assets.babylonjs.com/meshes/Tests/LevelOfDetail/", "LevelOfDetail.gltf", window.scene).then(() => {
+                    return window.scene!.whenReadyAsync().then(() => {
+                        return Promise.all(promises).then(() => {
+                            return data;
+                        });
+                    });
+                });
+            }, materialNames);
+
+            expect(Object.keys(assertionData)).toHaveLength(3);
+            materialNames.forEach((name) => {
+                expect(assertionData[name]).toEqual([name, name]);
+            });
+        });
+
+        // This test was deliberately left commented out as it is not working as expected.

```

Comment:
@bghgary - I am not sure what was the purpose of this test, but the onReady function in the MSFT_lod extension generated promises that cannot be stopped, even if the loader is disposed (and null-ed...). So this test will continue failing until fixed. I left it commented out, but I would remove it if you say it is fine.

Reply:
We need this test. It's one of the only tests covering the MSFT_lod extension.

Reply:
The test right before it is also using MSFT_lod:

`Load LevelOfDetail with onMaterialLODsLoadedObservable`

Reply:
The onReady function of the extension needs to be revised for this test to pass correctly